scale/job/views.py

```diff
@@ -764,7 +778,43 @@ class JobExecutionsView(ListAPIView):
     queryset = JobExecution.objects.all()
     serializer_class = JobExecutionSerializer
 
-    def list(self, request):
+    def list(self, request, job_id=None):
+        """Gets job executions and their associated job_type id, name, and version
+
+        :param request: the HTTP GET request
+        :type request: :class:`rest_framework.request.Request`
+        :param job_id: The ID for the job.
+        :type job_id: int encoded as a str
+        :rtype: :class:`rest_framework.response.Response`
+        :returns: the HTTP response to send back to the user
+        """
+
+        # TODO: remove this check when REST API v5 is removed
+        if not job_id:
+            if request.version == 'v5':
+                return self.list_v5(request)
+            else:
+                raise Http404
+        else:
+            started = rest_util.parse_timestamp(request, 'started', required=False)
+            ended = rest_util.parse_timestamp(request, 'ended', required=False)
+            rest_util.check_time_range(started, ended)
+
+            statuses = rest_util.parse_string_list(request, 'status', required=False)
+            node_ids = rest_util.parse_int_list(request, 'node_id', required=False)
+
+            job_exes = JobExecution.objects.get_job_exes(job_id=job_id, started=started, ended=ended,
+                                                         statuses=statuses, node_ids=node_ids)
+
+            if job_exes.exists():
```

Comment:
exists() does a separate query, so we don't want this. Just pass job_exes to the paginator.